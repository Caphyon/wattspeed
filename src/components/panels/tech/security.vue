<template>
  <container>
    <template slot="header">
      <div v-if="!error" class="section__header--stats px2 py1">
        <security-grade :grade="grades.headersGrade" section="Headers"></security-grade>
        <security-grade :grade="grades.sslGrade" section="SSL"></security-grade>
      </div>
    </template>
    <div class="scrollable__container">
      <div>
        <h4 class="text--center text--muted mt0 mb1">Headers</h4>
        <security-headers-item v-for="(data, key) in this.initialHeadersData" :title="key" :data="data" :key="key"/>
      </div>
      <div>
        <h4 class="text--center text--muted mt0 mb1">SSL</h4>
        <security-ssl-item v-for="(message, key) in this.initialSSLData.errors" :key="key" :message="message" type="error"/>
        <security-ssl-item v-for="(message, key) in this.initialSSLData.warnings" :key="key" :message="message" type="warning"/>
        <security-ssl-item v-for="(message, key) in this.initialSSLData.success" :key="key" :message="message" type="success"/>
      </div>
    </div>
  </container>
</template>

<script>
import Vue from "vue";
import SecuritySection from "../../sections/security-section";
import Container from "./container";
import HeadersItem from "./items/security-headers-item";
import SSLItem from "./items/security-ssl-item";

Vue.component("container", Container);
Vue.component("security-headers-item", HeadersItem);
Vue.component("security-ssl-item", SSLItem);

export default {
  mixins: [SecuritySection],

  data() {
    return {
      initialHeadersData: [],
      initialSSLData: [],
      desc: "Security evaluates response headers and SSL to detect any vulnerabilities against common threats. Included in the report are a list of headers we look for and information on what they represent. SSL displays a short validation summary of certificates, protocols and ciphers.",
    };
  },
};
</script>