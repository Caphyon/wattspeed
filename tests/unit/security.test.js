import Vue from 'vue';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import security from '../../src/components/panels/tech/security.vue'; // name of your Vue component
import security_grade from '../../src/components/panels/tech/items/security-grade.vue';
import security_headers_item from '../../src/components/panels/tech/items/security-headers-item.vue';
import security_ssl_item from '../../src/components/panels/tech/items/security-ssl-item.vue';

global.EventBus = new Vue();
const localVue = createLocalVue();

const data = {
  securityData: {},
  grades: {
    headersGrade: {
      grade: 'D',
      colour: 'orange',
    },
    sslGrade: {
      grade: 'B',
      colour: 'yellow',
    },
  },
  initialHeadersData: {
    'Strict-Transport-Security': {
      'value': 'max-age=31536000; includeSubDomains',
      'set': true,
    },
    'Content-Security-Policy': {
      'value': 'NA',
      'set': false,
    },
    'X-Frame-Options': {
      'value': 'SAMEORIGIN',
      'set': true,
    },
    'X-Content-Type-Options': {
      'value': 'NA',
      'set': false,
    },
    'Referrer-Policy': {
      'value': 'NA',
      'set': false,
    },
    'Permissions-Policy': {
      'value': 'NA',
      'set': false,
    },
  },
  initialSSLData: {
    'errors': {},
    'warnings': {
      'weakCiphers': 'This server supports weak cipher suites',
      'weakProtocols': 'This server supports weak TLS protocols',
    },
    'success': {
      'chain': 'Certificate is trusted',
    },
  },
  loading: false,
};

const methods = {
  getSecurity: jest.fn(),
};

function createWrapper(data) {
  return shallowMount(security, {
    localVue,
    methods,
    data() {
      return data;
    },
  });
}

describe('security.vue', () => {
  it('Testing speed component', () => {
    const wrapper = createWrapper(data);

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.section__header--stats.px2.py1').exists()).toBe(true);
    expect(wrapper.findAll(security_grade).length).toBe(2);
    expect(wrapper.findAll('.scrollable__container').exists()).toBe(true);

    expect(wrapper.findAll('.text--center.text--muted.mt0.mb1').length).toBe(2);
    expect(wrapper.findAll(security_headers_item).length).toBe(6);
    expect(wrapper.findAll(security_ssl_item).length).toBe(3);
    wrapper.destroy();
  });

});
