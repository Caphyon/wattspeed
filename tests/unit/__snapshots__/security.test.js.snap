// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`security.vue Testing speed component 1`] = `
<container-stub>
  <template>
    <div
      class="section__header--stats px2 py1"
    >
      <security-grade-stub
        grade="[object Object]"
        section="Headers"
      />
       
      <security-grade-stub
        grade="[object Object]"
        section="SSL"
      />
    </div>
  </template>
   
  <div
    class="scrollable__container"
  >
    <div>
      <h4
        class="text--center text--muted mt0 mb1"
      >
        Headers
      </h4>
       
      <security-headers-item-stub
        data="[object Object]"
        title="Strict-Transport-Security"
      />
      <security-headers-item-stub
        data="[object Object]"
        title="Content-Security-Policy"
      />
      <security-headers-item-stub
        data="[object Object]"
        title="X-Frame-Options"
      />
      <security-headers-item-stub
        data="[object Object]"
        title="X-Content-Type-Options"
      />
      <security-headers-item-stub
        data="[object Object]"
        title="Referrer-Policy"
      />
      <security-headers-item-stub
        data="[object Object]"
        title="Permissions-Policy"
      />
    </div>
     
    <div>
      <h4
        class="text--center text--muted mt0 mb1"
      >
        SSL
      </h4>
        
      <security-ssl-item-stub
        message="This server supports weak cipher suites"
        type="warning"
      />
      <security-ssl-item-stub
        message="This server supports weak TLS protocols"
        type="warning"
      />
       
      <security-ssl-item-stub
        message="Certificate is trusted"
        type="success"
      />
    </div>
  </div>
</container-stub>
`;
